package-system/OpenMesh/openmesh-o3de.patch

```diff
@@ -0,0 +1,193 @@
++void newell_norm(
++    Normal& n, const Point& a, const Point& b)
++{
++    n[0] += static_cast<typename vector_traits<Normal>::value_type>(a[1] * b[2]);
++    n[1] += static_cast<typename vector_traits<Normal>::value_type>(a[2] * b[0]);
++    n[2] += static_cast<typename vector_traits<Normal>::value_type>(a[0] * b[1]);
++}
++// o3de change end
++
+ template <class Kernel>
+ typename PolyMeshT<Kernel>::Normal
+ PolyMeshT<Kernel>::calc_face_normal_impl(FaceHandle _fh, PointIs3DTag) const
+ {
+   assert(this->halfedge_handle(_fh).is_valid());
+   ConstFaceVertexIter fv_it(this->cfv_iter(_fh));
+-  
++
+   // Safeguard for 1-gons
+   if (!(++fv_it).is_valid()) return Normal(0, 0, 0);
+ 
+@@ -131,16 +142,13 @@ PolyMeshT<Kernel>::calc_face_normal_impl(FaceHandle _fh, PointIs3DTag) const
+     const Point a = this->point(*fv_it) - this->point(*fv_itn);
+     const Point b = this->point(*fv_it) + this->point(*fv_itn);
+ 
+-
+     // Due to traits, the value types of normals and points can be different.
+     // Therefore we cast them here.
+-    n[0] += static_cast<typename vector_traits<Normal>::value_type>(a[1] * b[2]);
+-    n[1] += static_cast<typename vector_traits<Normal>::value_type>(a[2] * b[0]);
+-    n[2] += static_cast<typename vector_traits<Normal>::value_type>(a[0] * b[1]);
++    newell_norm(n, a, b); // o3de change
+   }
+ 
+   const typename vector_traits<Normal>::value_type length = norm(n);
+-  
++
+   // The expression ((n *= (1.0/norm)),n) is used because the OpenSG
+   // vector class does not return self after component-wise
+   // self-multiplication with a scalar!!!
+@@ -161,7 +169,7 @@ PolyMeshT<Kernel>::calc_face_normal_impl(FaceHandle, PointIsNot3DTag) const
+   assert(false);
+ 
+   Normal normal;
+-  vectorize(normal,Scalar(0));
++  vectorize(normal, Scalar(0.0)); // o3de fix
+   return normal;
+ }
+ 
+@@ -237,7 +245,7 @@ PolyMeshT<Kernel>::calc_face_normal_impl(const Point&, const Point&, const Point
+   assert(false);
+ 
+   Normal normal;
+-  vectorize(normal,Scalar(0));
++  vectorize(normal, Scalar(0.0)); // o3de fix
+   return normal;
+ }
+ 
+@@ -249,7 +257,7 @@ PolyMeshT<Kernel>::
+ calc_face_centroid(FaceHandle _fh) const
+ {
+   Point _pt;
+-  vectorize(_pt, Scalar(0));
++  vectorize(_pt, Scalar(0.0)); // o3de fix
+   Scalar valence = 0.0;
+   for (ConstFaceVertexIter cfv_it = this->cfv_iter(_fh); cfv_it.is_valid(); ++cfv_it, valence += 1.0)
+   {
+@@ -474,7 +482,7 @@ template <class Kernel>
+ void PolyMeshT<Kernel>::
+ calc_vertex_normal_fast(VertexHandle _vh, Normal& _n) const
+ {
+-  vectorize(_n, Scalar(0));
++  vectorize(_n, Scalar(0.0)); // o3de fix
+   for (ConstVertexFaceIter vf_it = this->cvf_iter(_vh); vf_it.is_valid(); ++vf_it)
+     _n += this->normal(*vf_it);
+ }
+@@ -484,7 +492,7 @@ template <class Kernel>
+ void PolyMeshT<Kernel>::
+ calc_vertex_normal_correct(VertexHandle _vh, Normal& _n) const
+ {
+-  vectorize(_n, Scalar(0));
++  vectorize(_n, Scalar(0.0)); // o3de fix
```

Comment:
suggestion: It looks like we might actually be able to remove these diffs, in OpenMesh 8.0 there was no cast, which is when we added it, but I think this was fixed in 8.1, our only change is making it `0.0` which isn't strictly required, so we could just revert these lines and keep vanilla OpenMesh

Reply:
Yeah, maybe for the next upgrade. It takes about half a day to go through the build and testing for all 3 platforms whenever I update the package, and I don't this will really affect the final binary much. But minimizing/eliminating our patches eventually is desirable

Reply:
Sure thing, totally makes sense :+1: